File: learning_c/learning_c/test/caesar.py
import readline
import sys


plain = input("Please input your plain text: ")
value = int(input("Please input your key (included negatives): "))

for letter in plain:
    if ord(letter) >= ord('A') and ord(letter) <= ord('Z'):
        print("%s" % chr(((ord(letter) - ord('A') + value) % 26 + ord('A'))), end='', file=sys.stderr)
    elif ord(letter) >= ord('a') and ord(letter) <= ord('z'):
        print("%s" % chr(((ord(letter) - ord('a') + value) % 26 + ord('a'))), end='', file=sys.stderr)
    else:
        print("%s" % letter, end='', file=sys.stderr)

print("", file=sys.stderr)

File: learning_c/learning_c/test/caesar.c
#include <stdio.h>
#include <string.h>
#define MAX 99902

int main(void) {
    char plain[MAX];
    char letter;
    int value;
    int index;

    printf("Please input your plain text: ");
    fgets(plain, MAX, stdin);
    printf("Please input your key (included negatives): ");
    scanf("%i", &value);

    for (index = 0; index < strlen(plain); index++) {
        letter = plain[index];

        if (letter >= 'A' && letter <= 'Z') {
            fprintf(stderr, "%c", (letter - 'A' + value) % 26 + 'A');
        }
                
        else if (letter >= 'a' && letter <= 'z') {
            fprintf(stderr, "%c", (letter - 'a' + value) % 26 + 'a');
        }
        else {
            fprintf(stderr, "%c", letter);
        }
    }
}
